ДЗ по Java #1

Sovianum opened the pull request.

KrylovS committed:
first commit

KrylovS committed:
Fixed UserProfile, fixed UserController.update and AccountService.update

Commit:
added wrong travis

Commit:
Merge branch 'develop' of https://github.com/KrylovS/ping-pong-02-2017 into develop

KrylovS committed:
Added HTTP status codes

Commit:
Another key 1

Commit:
Another key 2

Commit:
added gitignore

KrylovS committed:
Merge branch 'develop' of https://github.com/KrylovS/ping-pong-02-2017 into develop

Commit:
Cosmetic refactoring

Commit:
Another key 3

Commit:
Another key 4

Commit:
Another key 5

Commit:
Another key 6

Commit:
Another key 7

Commit:
Added API description

Commit:
updated README.md

isopov commented:
1. gitignore - какая-то каша, почистите, пожалуйста
2. travis.yml - куча непонятного мусора, нужно комментировать, или это бездумная копипаста откуда-то? ну или если уже подключили travis (мододцы!) - сделайте на него ссылку из readme
3. зачем подключать и использовать эту гадость? Чем Jackson не устроил?
```
<dependency>
    <groupId>org.json</groupId>
    <artifactId>json</artifactId>
    <version>20160212</version>
</dependency>
```
4."userNameToUserProfile - хранилище пользователей, в качестве ключа используется email" - комментарии нужны для неочевидного кода. Здесь неочивдность только в неправильном названии переменной. лучше сделать правильное название, а комментарий удалить.
5. `@RequestMapping(path = "/api/user/login", method = RequestMethod.POST, produces = "application/json", consumes = "application/json")` -> '@PostMapping("/api/user/login")'

Прошу прощения за задержку.

Commit:
1. Got rid of rubbish in .gitignore;
2. Got rid of redundant dependency;
3. UserNameToProfile => userStorage;
4. Simplified mappers' anotations.

Commit:
1. Added travis to README.md
2. Commented .travis.yml
3. Simplified .travis.yml (configuration not tested yet)

Commit:
Specified jdk version cos travis defauld jdk fails trying to compile to java8.

Commit:
Moved to jackson

Sovianum commented:
1. gitignore почистил.
2. В travis ненужные опции убрал, добавил комменты. 
3. Зависимость выпилил, заменил использованные объекты аналогами из jackson.
4. Коммент удалил, переименовал userNameToUserProfile на userStorage.
5. Анотации упростил.

KSolovyev commented:
@isopov Иван?

Sovianum commented:
Извините, пожалуйста, не могли бы Вы написать моменты, которые нужно переделать в нашем проекте?

isopov commented:
Прошу прощения. Итак. Задавайте вопрос в чат, если что-то непонятно, мне казалось, что я объяснил в чем были проблемы на РК, но оказывается, не сумел:
1. Удалите всё такое:
```
         final ObjectMapper mapper = new ObjectMapper();
        ObjectNode responseJSON = mapper.createObjectNode();
        final ArrayNode errorList = mapper.createArrayNode();
```
Вы принимаете json с клиента просто в объект, так и отдавайте просто объект, и пусть спринг его джексоном превратит в json.
2. В чем смысл такого метода? (Мусор, затрудняющий чтение)
```
    private boolean isOccupied(String key) {
        return userStorage.containsKey(key);
    }
```
От любого метода при чтении ожидаешь какой-то логики, а не единственного вызова другого метода. Единственный вызов может быть для реализации интерфейса, работы с внешним API и т.д. Здесь он явно не нужен. Удалите и будет явно видно, что вы перед get-ом проверяете карту через containsKey - не надо делать эту двойную работу. Делайте просто get, и проверяйте результат на null.

KrylovS committed:
Added some methods for score and fixed UserController and AccountService
